src/SlackGhost.ts

```diff
@@ -0,0 +1,324 @@
+                return existingReq.user;
+            }
+            return;
+        }
+        log.debug("Using fresh userInfo for", slackUserId);
+
+        this.main.incRemoteCallCounter("users.info");
+        this.userInfoLoading = rp({
+            json: true,
+            qs: {
+                token: slackAccessToken,
+                user: slackUserId,
+            },
+            uri: "https://slack.com/api/users.info",
+        }) as rp.RequestPromise<{user?: ISlackUser}>;
+        const response = await this.userInfoLoading!;
+        if (!response.user || !response.user.profile) {
+            log.error("Failed to get user profile", response);
+            return;
+        }
+        this.userInfoCache = response.user;
+        setTimeout(() => this.userInfoCache = undefined, USER_CACHE_TIMEOUT);
+        this.userInfoLoading = undefined;
+        return response.user!;
+    }
+
+    public async updateAvatar(message: {bot_id?: string, user_id?: string}, room: BridgedRoom) {
+        const token = room.AccessToken;
+        if (!token) {
+            return;
+        }
+
+        let avatarUrl;
+        if (message.bot_id) {
+            avatarUrl = await this.getBotAvatarUrl(message.bot_id, token);
+        } else if (message.user_id) {
+            avatarUrl = await this.lookupAvatarUrl(message.user_id, token);
+        } else {
+            return;
+        }
+
+        if (this.avatarUrl === avatarUrl) {
+            return;
+        }
+
+        const match = avatarUrl.match(/\/([^\/]+)$/);
+        if (!match || !match[1]) {
+            return;
+        }
+
+        const title = match[1];
+
+        const response = await rp({
+            encoding: null,
+            resolveWithFullResponse: true,
+            uri: avatarUrl,
+        });
+        const contentUri = await this.uploadContent({
+            _content: response.body,
+            mimetype: response.headers["content-type"],
+            title,
+        });
+        await this.intent.setAvatarUrl(contentUri);
+        this.avatarUrl = avatarUrl;
+        this.main.putUserToStore(this);
+    }
+
+    public prepareBody(body: string) {
+        // TODO: This is fixing plaintext mentions, but should be refactored. See issue #110
+        return body.replace(/<https:\/\/matrix\.to\/#\/@.+:.+\|(.+)>/g, "$1");
+    }
+
+    public prepareFormattedBody(body: string) {
+        return Slackdown.parse(body);
+    }
+
+    public async sendText(roomId: string, text: string, slackRoomID: string, slackEventTS: string, extra: {} = {}) {
+        // TODO: Slack's markdown is their own thing that isn't really markdown,
+        // but the only parser we have for it is slackdown. However, Matrix expects
+        // a variant of markdown that is in the realm of sanity. Currently text
+        // will be slack's markdown until we've got a slack -> markdown parser.
+
+        // TODO: This is fixing plaintext mentions, but should be refactored. See issue #110
```

Comment:
Can we replace "See issue #110" with a URL please? (applies to other comments too).